docs/standard/serialization/system-text-json/populate-properties.md

```diff
@@ -0,0 +1,62 @@
+---
+title: Populate initialized properties
+description: "Learn how to modify properties and fields that are already initialized when deserializing from JSON in .NET."
+ms.date: 10/20/2023
+no-loc: [System.Text.Json, Newtonsoft.Json]
+dev_langs:
+  - "csharp"
+ms.topic: how-to
+---
+
+# Populate initialized properties
+
+The System.Text.Json deserializer always creates a new instance of the target type. However, even though a new instance is created, some properties and fields might already be initialized as part of the object's construction. Consider the following type:
+
+```csharp
+class A
+{
+    public List<int> Numbers1 { get; } = new List<int>() { 1, 2, 3 };
+    public List<int> Numbers2 { get; set; } = new List<int>() { 1, 2, 3 };
+}
+```
+
+When you create an instance of this class, the `Numbers1` (and `Numbers2`) property's value is a list with three elements (1, 2, and 3). If you deserialize JSON to this type, the default behavior is that property values are replaced:
+
+- For `Numbers1`, since it's read-only (no setter), it will still have the values 1, 2, and 3 in its list.
+- For `Numbers2`, which is read-write, the contents of the list will be replaced with the values from the JSON.
+
+For example, if you execute the following deserialization code, `Numbers1` contains the values 1, 2, and 3 and `Numbers2` contains the values 4, 5, and 6.
```

Comment:
In this example, I'm curious if the `List` object is replaced with a newly allocated list, or if the current list is cleared and new items added. 

I can think of (mostly evil) code examples that would allow the difference to be observed. (Like grabbing a reference to the `Numbers2` property in a constructor and storing it in a different field.

Reply:
Based on what I know, I think it's a newly allocated list. Either way I don't think it needs to go in this doc.

Reply:
The default `Replace` behavior will always allocate a new list. The `Populate` behavior will reuse without clearing so if the collection is pre-populated with elements they will show in the final deserialized result. Perhaps that might be worth mentioning here.